src/esaml_cowboy.erl

```diff
@@ -210,6 +198,14 @@ perform_extra_security_if_applicable(Callback,   Callback_State,  Xml, Assertion
         {error, E}  -> {error, E, Req}
     end.
 
+qs_val(Key, Req) when is_atom(Key) ->
```

Comment:
I'd just remove the Req from the response altogether instead of passing it back & forth